src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,13 +248,28 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
-        Trace process("HidController process packet");
-        QByteArray outData(reinterpret_cast<char*>(m_pPollData), result);
-        receive(outData, mixxx::Time::elapsed());
+    int result = 1;
+    while (result > 0) {
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+        m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+
+        result = hid_read(m_pHidDevice, pCurrentBuffer, kBufferSize);
+        if (result == -1) {
+            return false;
+        } else if (result > 0) {
+            Trace process("HidController process packet");
+            // Some controllers such as the Gemini GMX continuously send input packets even if it
+            // is identical to the previous packet. If this loop processed all those redundant
+            // packets, it would be a big performance problem.
+            if (memcmp(pCurrentBuffer, pPreviousBuffer, kBufferSize) == 0) {
+                continue;
```

Comment:
If we decide not to make this code conditional, we need a more significant comment that a HID telegram is discarded here for all controllers anong with some words why we think that his is universal correct.

Reply:
I disagree that there is any need for further explanation in the comment. If the current packet is the same as the last, there is no purpose processing it, regardless of which controller is doing this.

Reply:
Additional comments never hurts. It is better to be explicit. 
The long discussion shows that there is a need IMHO.

Reply:
The code is IMO obvious and explicit already.

Reply:
This is normal for the author...

Reply:
Comments should never describe exactly what the code is doing, e.g.:

`// Add one
++i;`

This particular comment is borderline -- it's obvious what the code does but it helps to know the context.  How about: "// Some controllers send many redundant packets.  Be sure to ignore duplicates."  Three lines is too verbose.

"            // Some controllers such as the Gemini GMX continuously send input packets even if it
            // is identical to the previous packet. If this loop processed all those redundant
            // packets, it would be a big performance problem.

Reply:
No, that's too terse. It does not explain why duplicates are a problem. There is nothing wrong with a 3 line comment.

Reply:
Please add this in addition to your comment:

```
// This loop discards all equal HID packages. This releases the js mapping code from idle work in case the controller repeat it's complete process state in a short loop. Similar checks in that mapping itself can be omitted.
We consider this as universal correct, because a protocol with different telegrams will likely have different content. The same is true for an event driven protocol.
This loop has the risk to be an high priority endless loop in case processing the mapping code takes longer than the controller update rate. This may stall other low priority task.
We have no safety net here, because we consider this as very unlikely.
```

Reply:
done 447a432

Reply:
Are the comments sufficient?

Reply:
ping